share/availability/light/availability.go

```diff
@@ -113,10 +113,10 @@ func (la *ShareAvailability) SharesAvailable(ctx context.Context, header *header
 	}
 
 	smpls, err := la.getter.GetSamples(ctx, header, idxs)
-	if errors.Is(ctx.Err(), context.Canceled) {
+	if errors.Is(err, context.Canceled) {
```

Comment:
Better to check `err` rather than `ctx.Err`

Reply:
Btw, we will have to rebase on recent changes from Vlad here

Reply:
Well, this PR and branch are targeting your PR. So, firstly this PR should be merged into yours and then rebased on Vlad's changes.

Reply:
I didn't mean this PR, but this code here